pkg/vz/vm_darwin.go

```diff
@@ -686,6 +681,40 @@ func getMachineIdentifier(driver *driver.BaseDriver) (*vz.GenericMachineIdentifi
 	return vz.NewGenericMachineIdentifierWithDataPath(identifier)
 }
 
+func bootLoader(driver *driver.BaseDriver) (vz.BootLoader, error) {
+	if linuxBootLoder, err := linuxBootLoader(driver); linuxBootLoder != nil {
+		return linuxBootLoder, err
+	}
+
+	efiVariableStore, err := getEFI(driver)
+	if err != nil {
+		return nil, err
+	}
+	logrus.Debugf("Using EFI Boot Loader")
+	return vz.NewEFIBootLoader(vz.WithEFIVariableStore(efiVariableStore))
+}
+
+func linuxBootLoader(driver *driver.BaseDriver) (*vz.LinuxBootLoader, error) {
```

Comment:
Please add a comment to clarify that this function may return nil without an error

Reply:
changed `linuxBootLoader()` to return error, then explicitly ignores the error on caller.